sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/NexmarkLauncher.java

```diff
@@ -1154,4 +1140,81 @@ public NexmarkPerf run(NexmarkConfiguration runConfiguration) {
       queryName = null;
     }
   }
+
+  private boolean isSql() {
+    return SQL.equalsIgnoreCase(options.getQueryLanguage());
+  }
+
+  private NexmarkQueryModel getNexmarkQueryModel() {
+    List<NexmarkQueryModel> models = createQueryModels();
+    return models.get(configuration.query);
+  }
+
+  private NexmarkQuery getNexmarkQuery() {
+    List<NexmarkQuery> queries = createQueries();
+
+    if (options.getQuery() >= queries.size()) {
+      throw new UnsupportedOperationException(
+          "Query " + options.getQuery()
+          + " is not implemented yet");
+    }
+
+    return queries.get(configuration.query);
+  }
+
+  private List<NexmarkQueryModel> createQueryModels() {
+    return isSql()
+        ? createSqlQueryModels()
+        : createJavaQueryModels();
+  }
+
+  private List<NexmarkQueryModel> createSqlQueryModels() {
+    return Arrays.asList(
+        null, null, null, null, null, null, null, null, null, null, null, null);
+  }
+
+  private List<NexmarkQueryModel> createJavaQueryModels() {
+    return Arrays.asList(
```

Comment:
<!--new_thread; commit:964cc527e15c17e26b9584aaaf993cddc225398b; resolved:1-->
Nit: `Immutable.list.of(... elems ...)` is a little more readable than `Arrays.asList`.